blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticIO.java

```diff
@@ -150,20 +149,23 @@ private IndexRequestBuilder createSingleRequest(Discovery md) throws IOException
         
         List<SearchResult> results = new ArrayList<SearchResult>();
         Timer.Context searchTimerCtx = searchTimer.time();
-        
-        // todo: we'll want to change this once we decide and use a query syntax in the query string.
-        BoolQueryBuilder qb = boolQuery()
+        QueryBuilder qb;
+
+        GlobPattern pattern = new GlobPattern(query);
+        if (pattern.hasWildcard()) {
+            qb = wildcardQuery(metric_name.name(), query);
```

Comment:
Yes, you were right! Changed it to support term queries for non-wildcard characters and regexp for wildcard ones in 7c0a3b1